### PR TITLE
Drop react-a11y rules

### PR DESCRIPTION
**Pull request by ockham**
Consumers can just use upstream's `plugin:jsx-a11y/recommended` ruleset instead.

Prior discussion: https://github.com/Automattic/wp-calypso/pull/21687#discussion_r162668127

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#### v2.0.0 (2018-01-19)
+
+- Breaking: Removed the `react-a11y` ruleset. Consumers can just use upstream's `plugin:jsx-a11y/recommended` ruleset instead.
```
[the remark this answers is not in the document]
Comment (sirreal) [reply CHANGELOG.md]: I think it's a good thing to keep in mind 🙂

File: react-a11y/index.js [hunk 2]
```diff
@@ -1,37 +0,0 @@
-module.exports = {
-	'extends': '../react/index.js',
```
Comment (sirreal) [inline react-a11y/index.js]: Would it be preferable and feasible to add an extends rule for [react-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y#usage)?

```js
{
  'extends': [
    '../react/index.js',
    'plugin:jsx-a11y/recommended'
  ]
}
```

That would seem to strike a good balance as consumers wouldn't need to add the extends themselves and we could stop maintaining a ruleset ourselves.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ockham) [reply CHANGELOG.md]: Yep, good point! 0544891

File: react-a11y/index.js (hunk 2 above)
Comment (ockham) [reply react-a11y/index.js]: See https://github.com/Automattic/wp-calypso/pull/21687#discussion_r162689099 for some musings on that. I don't have a very strong opinion on this, it just seemed to make sense to separate concerns and leave it up to consumers to import the a11y ruleset themselves (much like Calypso also [imports the recommended jest ruleset directly](https://github.com/Automattic/wp-calypso/blob/6812e4b9ed2cbf65cf7ecaba2ff5fcb75e37865b/.eslintrc.js#L5)).

**Comment by ockham** [comment]
Thanks a lot for researching other users @ryelle and for reviewing @sirreal! Going to merge & publish.

**Pull request merged**